deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Clusters.java

```diff
@@ -124,6 +133,38 @@ public void deleteCluster(@Context SecurityContext sc,
     }
 
     @PUT
+    @Path("/actions")
+    public void replaceCluster(@Context SecurityContext sc,
+                               @PathParam("envName") String envName,
+                               @PathParam("stageName") String stageName,
+                               @QueryParam("type") String type) throws Exception {
```

Comment:
We should use ActionType directly - the framework will be responsible to convert string to enum automatically for us, and throw exceptions if string is illegal.
@QueryParam("type") ActionType type